plugin/adsk/scripts/mayaUsd_USDRootFileRelative.py

```diff
@@ -110,9 +161,100 @@ def uiCommit(cls, parentLayout, relativeToWhat):
         cmds.setParent(parentLayout)
 
         # Get the current checkbox state and save to optionVar.
-        relative = cmds.checkBox(cls.kMakePathRelativeCheckBox + relativeToWhat, query=True, value=True)
+        relative = cmds.checkBox(cls.kMakePathRelativeCheckBox, query=True, value=True)
         cmds.optionVar(iv=('mayaUsd_MakePathRelativeTo' + relativeToWhat, relative))
 
+    @classmethod
+    def connectToDialogControls(cls, parentLayout):
+        """
+        Find some of the conrols (such as file name edit field) in the dialog
+        (under the input parent layout) and connect to them for change notifs.
+        Used so we can update the file path preview fields.
+        """
+
+        # Get the Qt pointer for the input parent string (layout from mel).
+        ptr = omui.MQtUtil.findLayout(parentLayout)
+        if ptr is not None:
+            # Find the top-level window from that parent layout.
+            maya_widget = wrapInstance(int(ptr), QWidget)
+            pp = maya_widget.parent()
+            while pp:
+                if pp.inherits('QFileDialog'):
+                    cls._fileDialog = pp
+                pp = pp.parent()
+                if pp:
+                    maya_window = pp
+
+        if maya_window:
+            # Find the file name edit field and connect to it to be notified when text changes.
+            fileNameEditField = maya_window.findChild(QLineEdit, 'fileNameEdit')
+            if fileNameEditField:
+                fileNameEditField.textChanged.connect(cls.onfileNameEditFieldChanged)
+
+            # Find the accept button, so we can know when there is valid input in the file name edit field.
+            buttonBox = maya_window.findChild(QDialogButtonBox, 'buttonBox')
+            if buttonBox and cls._fileDialog:
+                btn = QDialogButtonBox.Open if (cls._fileDialog.acceptMode() == QFileDialog.AcceptOpen) else QDialogButtonBox.Save
+                cls._acceptButton = buttonBox.button(btn)
```

Comment:
The dialog will enable/disable the OK button according to whether or not there is valid filename. So I use the state of that button to know if there is a valid filename to query.